### DISCUSSION TITLE
2D Mesh Extrusion

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#28300] [Q&A Meshing] [answered]
Is it possible to create a AnnularMeshGenerator generated mesh with differing mesh bias at the ID and OD of the geometry? If so can that 2D mesh be extruded with bias from the origin that decreases (decreased mesh growth} with each subsequent level of mesh height?

**Comment by GiudGiud** [comment, marked as answer]
Hello

The AdvancedMeshExtruder will let you extrude with different axial layer thicknesses. 

There are numerous circular mesh generator. I'm sure there is one with a radial biasing option

Guillaume

**Reply by GiudGiud** [reply]
https://mooseframework.inl.gov/source/meshgenerators/PolygonConcentricCircleMeshGenerator.html will let you pick ring sizes.
Just delete the regions outside the cylinder using a BlockDeletionGenerator after

**Reply by TLWise** [reply]
Thank you. I was able to do what I had asked by using two instances of AnnularMeshGenerator, one to make an outer ring with negative bias and the other to make an inner ring with positive bias. I then used StitchedMeshGenerator to combine the two coincident surfaces, removing the previous surface names. I extruded the mesh using AdvancedExtruderGenerator, with bias.

**Reply by GiudGiud** [reply]
I'm glad. Thanks for reporting back about it!
